app/controllers/changeContactDetails/ChangeOrganisationContactDetailsController.scala

```diff
@@ -0,0 +1,103 @@
+    with I18nSupport
+    with Logging {
+
+  private val allowedAffinityGroups: Set[AffinityGroup] = Set(AffinityGroup.Organisation, AffinityGroup.Agent)
+
+  def onPageLoad: Action[AnyContent] = standardActionSets.subscriptionIdWithChangeDetailsRetrieval(allowedAffinityGroups).async {
+    implicit request =>
+      if (request.userAnswers.get(ChangeContactDetailsInProgressPage).isEmpty) {
+        subscriptionService.getContactDetails(request.subscriptionId, request.userAnswers).flatMap {
+          case Some(userAnswers) =>
+            (for {
+              updatedUserAnswers <- Future.fromTry(userAnswers.set(ChangeContactDetailsInProgressPage, true))
+              _                  <- sessionRepository.set(updatedUserAnswers)
+            } yield {
+              val helper               = new ChangeOrganisationContactDetailsHelper(updatedUserAnswers, countryFactory)
+              val primaryContactList   = SummaryListViewModel(helper.changeOrganisationPrimaryContactDetails)
+              val secondaryContactList = SummaryListViewModel(helper.changeOrganisationSecondaryContactDetails)
+
+              subscriptionService.contactDetailsChanged(request.subscriptionId, updatedUserAnswers) flatMap {
+                case Some(hasChanges) => Future.successful(Ok(view(primaryContactList, secondaryContactList, frontendAppConfig, hasChanges)))
+                case _                => Future.successful(InternalServerError(errorView()))
+              }
+            }).flatten
+          case _ =>
+            logger.warn(s"Failed to retrieve subscription for ${request.subscriptionId.value}")
+            Future.successful(InternalServerError(errorView()))
+        }
+      } else {
+        val helper               = new ChangeOrganisationContactDetailsHelper(request.userAnswers, countryFactory)
+        val primaryContactList   = SummaryListViewModel(helper.changeOrganisationPrimaryContactDetails)
+        val secondaryContactList = SummaryListViewModel(helper.changeOrganisationSecondaryContactDetails)
+        subscriptionService.contactDetailsChanged(request.subscriptionId, request.userAnswers) map {
+          case Some(hasChanges) => Ok(view(primaryContactList, secondaryContactList, frontendAppConfig, hasChanges))
+          case _                => InternalServerError(errorView())
+        }
+      }
+  }
+
+  def onSubmit: Action[AnyContent] = standardActionSets.subscriptionIdWithChangeDetailsRetrieval(allowedAffinityGroups).async {
+    implicit request =>
+      subscriptionService.updateContactDetails(request.subscriptionId, request.userAnswers) map {
+        case true =>
+          request.userAnswers.remove(ChangeContactDetailsInProgressPage) match {
+            case Success(_) =>
+              Redirect(controllers.routes.DetailsUpdatedController.onPageLoad())
+            case Failure(exception) =>
+              logger.error(s"Failed to remove $ChangeContactDetailsInProgressPage", exception)
```

Comment:
Should this be a `error` log or a `warn` log? As per guidance:

**Error** - for significant issues that you will cause someone to be woken up at night to fix, such as inability to perform the service's central responsibility. These should very rarely happen. An example is an OutOfMemoryError or IO error.

**Warn** - for significant issues that will get someone's attention. Examples include unexpected data returned from a dependent service.